### PR TITLE
global: migration from bower

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

File: invenio_assets/bower.py [hunk 1]
```diff
@@ -28,5 +43,6 @@ def __init__(self, *contents, **options):
         :param bower: bower dependencies
         :type bower: dict
         """
-        self.bower = options.pop("bower", {})
-        super(BowerBundle, self).__init__(*contents, **options)
+        raise DeprecationWarning(
+            "`Bower` has been deprecated please migrate to `npm`."
+        )
```
Comment (drjova) [inline invenio_assets/bower.py]: @drjova add better warning

File: invenio_assets/cli.py [hunk 2]
```diff
@@ -53,13 +53,13 @@ def _webassets_cmd(cmd):
```
Comment (drjova) [inline invenio_assets/cli.py]: @drjova Maybe keep the bower command with a deprecation message?

**Comment by drjova** [comment]
@greut any idea how to solve dependencies transparently? :)

File: tests/test_npm.py [hunk 3]
```diff
@@ -96,12 +96,12 @@ def test_cli(script_info):
             f.write(json.dumps({'dependencies': newdep}))
 
         result = runner.invoke(
-            bower, ['-i', 'base.json', '-o', 'bower.json'], obj=script_info)
+            npm, ['-i', 'base.json', '-o', 'package.json'], obj=script_info)
```
Comment (greut) [inline tests/test_npm.py]: IMHO, the `-i` option can go away.

File: invenio_assets/bower.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_assets/bower.py]: can we transparently create `NpmBundle`:
```python
options['npm'] = options.pop('bower', {})
super(NpmBundle, self).__init__(*contents, **options)
```
+ the warning of course?

Comment (lnielsen) [inline invenio_assets/bower.py]: Can't we just remove the bower bundle? We're still in alpha and no promises have been given, so rather get it out right away before it's released.

File: invenio_assets/cli.py (hunk 2 above)
Comment (lnielsen) [inline invenio_assets/cli.py]: IMHO We don't need it :-)

File: tests/test_npm.py (hunk 3 above)
Comment (lnielsen) [inline tests/test_npm.py]: Yep, agree.

**Comment by lnielsen** [comment]
Note that Invenio-Accounts, Invenio-Theme, and Invenio have documentation that needs to be updated after this PR has been integrated.

File: invenio_assets/bower.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_assets/bower.py]: I think it's good practice even in "alpha world" to warn users if you can.

**Comment by jirikuncar** [comment]
> Note that Invenio-Accounts, Invenio-Theme, and Invenio have documentation that needs to be updated after this PR has been integrated.

That's why having a little compatibility layer that we can remove in release-candidate can improve the user experience with Invenio 3.0 alpha world.

**Comment by lnielsen** [comment]
> > Note that Invenio-Accounts, Invenio-Theme, and Invenio have documentation that needs to be updated after this PR has been integrated.

> That's why having a little compatibility layer that we can remove in release-candidate can improve the user experience with Invenio 3.0 alpha world.

But then we also need to keep the ``bower`` CLI command around as well. IMHO it's not worth the effort. Rather get the fix done fast as well as dependent packages updated and release at the same time, than keeping open issues and "dirty code" around.

**Comment by tiborsimko** [comment]
:+1: for "not worth the effort" regarding the compatibility layer.  

Early adopters of v3.0 alpha should be ready to read the tickets and git logs and adjust in no time :) We can distribute the work among "sprinters" in order to update related packages and docs and kickstart scripts as the new alpha packages are being pushed to PyPI.

See also musings at https://github.com/inveniosoftware/invenio/pull/3566#discussion_r45033404

**Comment by drjova** [comment]
https://github.com/inveniosoftware/invenio-theme/pull/22

File: invenio_assets/__init__.py [hunk 4]
```diff
@@ -67,12 +67,12 @@
 
 from __future__ import absolute_import, print_function
 
-from .bower import BowerBundle
+from .npm import NpmBundle
```
Comment (jirikuncar) [inline invenio_assets/__init__.py]: it probably needs resorting

File: invenio_assets/__init__.py [hunk 5]
```diff
@@ -67,12 +67,12 @@
...
 from .ext import InvenioAssets
 from .filters import CleanCSSFilter, RequireJSFilter
 from .proxy import current_assets
 from .version import __version__
 
 __all__ = (
-    '__version__', 'BowerBundle', 'CleanCSSFilter', 'InvenioAssets',
+    '__version__', 'NpmBundle', 'CleanCSSFilter', 'InvenioAssets',
```
Comment (jirikuncar) [inline invenio_assets/__init__.py]: https://github.com/inveniosoftware/invenio-assets/pull/11/files#r45041071

**Comment by greut** [comment]
@drjova the prefix is a good workaround for the static files.

I continue to dislike the idea that npm is present on the deployment server though.

File: invenio_assets/cli.py [hunk 6]
```diff
@@ -52,14 +53,25 @@ def _webassets_cmd(cmd):
     getattr(cmdenv, cmd)()
 
 
+def _extract_deps(bundle):
```
Comment (jirikuncar) [inline invenio_assets/cli.py]: It should probably go to `npm.py`

Comment (greut) [inline invenio_assets/cli.py]: k, I'll move those in drjova/invenio-assets#2.

**Comment by jirikuncar** [comment]
@drjova please add an `INCOMPATIBLE` label with short description how to migrate (e.g. amend import, rename bower property to npm and use `npm` command to generate `package.json`) otherwise we can :shipit:

**Comment by greut** [comment]
@drjova I did get the chance to fix this one...
```diff
+++ invenio_assets/npm.py
 import semver
-
 from flask_assets import Bundle as BundleBase
```
https://travis-ci.org/drjova/invenio-assets/jobs/91627103

Sorry that you have to amend this...

**Comment by drjova** [comment]
:+1:

**Pull request merged**